### PR TITLE
App context changes

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
These changes are usable with DTR app context changes.  This doesn't really have much to do with relaunching other than modification of the Questionnaire operation to function properly when given an ID parameter.  

To test these changes, you can check the app context when loading DTR to find that it is conformant to the new IG standard, removing filepath and fhirpath and introducing coverage/response.  

Overall the major changes are:

- Modification of the smartLinkBuilder to produce the new appContext (CdsService)
- The introduction of an ID specific questionnaire-package endpoint to handle that case (FhirController) 
- Modification of the questionnaire-package operation to work with ID specific requests (QuestionnairePackageOperation)

### CODE REVIEW

**Comment by KeeyanGhoreshi** [comment]
Merging master, it seems like one of the commits from the deidentification pr is causing a bug, I'm looking into it.

File: server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java [hunk 1]
```diff
@@ -385,11 +370,8 @@ private Link smartLinkBuilder(String patientId, String fhirBase, URL application
     // request is the ID of the device request or medrec (not the full URI like the
     // IG says, since it should be taken from fhirBase
 
-    String appContext = "template=" + questionnaireUri + "&request=" + reqResourceId;
-    appContext = appContext + "&fhirpath=" + applicationBaseUrl + "/fhir/";
+    String appContext = "questionnaire=" + questionnaireUri + "&order=" + reqResourceId + "&coverage=" + FhirRequestProcessor.getCoverageFromRequest(request).getReference();
```
Comment (yuliumitre) [inline server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java:373]: Does the appContext still need coverage with the latest BR call discussion?

File: server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java [hunk 2]
```diff
@@ -385,11 +370,8 @@ private Link smartLinkBuilder(String patientId, String fhirBase, URL application
     // request is the ID of the device request or medrec (not the full URI like the
     // IG says, since it should be taken from fhirBase
```
Comment (yuliumitre) [inline server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java:372]: Please update the comments for the PARAMS here.

File: server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java (hunk 1 above)
Comment (KeeyanGhoreshi) [reply server/src/main/java/org/hl7/davinci/endpoint/cdshooks/services/crd/CdsService.java:373]: its still needed for the questionnairePackage operation, unless I missed something in the latest BR call

**Comment by yuliumitre** [comment]
@KeeyanGhoreshi I got 500 error (Nullpointer) when trying to call the interface without the QuestionnaireID.  http://localhost:8090/fhir/r4/Questionnaire/$questionnaire-package 
We can create a JIRA ticket to track the issue.

**Comment by yuliumitre** [comment]
@KeeyanGhoreshi can you resolve the conflicts pls? Thanks.

**Review by plarocque4 — COMMENTED** [review]
Looks good once the conflicts are resolved.

**Review by plarocque4 — APPROVED** [review]
Works well.

**Pull request merged**